segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -265,6 +262,45 @@ public void close() {
         return segment.append(s, null, timeout);
     }
 
+    private <T> CompletableFuture<AsyncIterator<IteratorItem<T>>> newIterator(@NonNull String segmentName, byte[] serializedState,
+                                                                              @NonNull Duration fetchTimeout,
+                                                                              @NonNull GetBucketReader<T> createBucketReader) {
+        UUID fromHash;
+        try {
+            fromHash = KeyHasher.getNextHash(serializedState == null ? null : IteratorState.deserialize(serializedState).getKeyHash());
```

Comment:
Instead of incrementing the hash, couldn't you just has an 'exclusive' flag?

Reply:
Yeah that would be ideal, however this makes use of two underlying APIs, one of which is an Attribute Iterator. For this, I'd have to change the Attribute Iterator to support such an API. I found it simpler to do it here, especially given the logic inside `getNextHash` which knows what kind of Attributes constitute hashes (It excludes Core Attributes (UUID.MSB==Long.MIN_VALUE) and Backpointers (UUID.MSB == Long.MAX_VALUE)). I would not want to add such information in the downstream code.